main.c

```diff
@@ -200,48 +286,331 @@ int main(void)
+        0x27, 0xae, 0x41, 0xe4, 0x64, 0x9b, 0x93, 0x4c,
+        0xa4, 0x95, 0x99, 0x1b, 0x78, 0x52, 0xb8, 0x55
+    };
 
-    ASSERT_SUCCESS_PSA(psa_import_key(verify_handle, key_type, pubkey,
-                                      pubkey_len));
+    ASSERT_SUCCESS_PSA(atecc608a_hash_sha256(hash_input1,
+                                             sizeof(hash_input1) - 1,
+                                             sha256_expected_hash1,
+                                             sizeof(sha256_expected_hash1)));
 
-    /* Verify that the signature produced by the secure element is valid. */
-    ASSERT_SUCCESS_PSA(psa_asymmetric_verify(verify_handle, alg, hash,
-                                             sizeof(hash), signature,
-                                             signature_length));
+    ASSERT_SUCCESS_PSA(atecc608a_hash_sha256(hash_input2,
+                                             sizeof(hash_input2) - 1,
+                                             sha256_expected_hash2,
+                                             sizeof(sha256_expected_hash2)));
+
+    printf("test_hash_sha256 succesful!\n");
+exit:
+    return status;
+}
+
+psa_status_t run_tests()
+{
+    psa_status_t status;
+
+    printf("Running tests...\n");
+    ASSERT_SUCCESS_PSA(test_hash_sha256());
+
+    /* Verify that the device has a locked config zone before running tests
+     * that use slots. */
+    ASSERT_SUCCESS_PSA(atecc608a_check_zone_locked(LOCK_ZONE_CONFIG));
+
+    ASSERT_SUCCESS_PSA(test_generate_import());
+    ASSERT_SUCCESS_PSA(test_export_import());
+    ASSERT_SUCCESS_PSA(test_sign_verify());
+    ASSERT_SUCCESS_PSA(test_psa_import_verify());
+
+    /* Verify that the device has a locked data zone before running tests
+     * that use clear text read. */
+    ASSERT_SUCCESS_PSA(atecc608a_check_zone_locked(LOCK_ZONE_DATA));
+
+    /* Slot 8 is usually used as a clear write and read certificate
+     * or signature slot, as it is the biggest one (416 bytes of space). */
+    ASSERT_SUCCESS_PSA(test_write_read_slot(8));
 
-    printf("Verification successful!\n");
     exit:
         return status;
 }
+
+void print_device_info()
+{
+    atecc608a_print_serial_number();
+    atecc608a_print_config_zone();
+    atecc608a_print_locked_zones();
+    printf("\nPrivate key slot in use: %lu, public: %lu\n",
+           atecc608a_private_key_slot, atecc608a_public_key_slot);
+}
+
+bool interactive_loop()
+{
+    char command [80];
+    char *arg;
+    size_t len;
+
+    printf(USAGE);
+    scanf ("%79s",command);
+    len = strlen(command);
+
+    arg = strchr( command, '=' );
+
+    if (strcmp(command, "info") == 0)
+    {
+        print_device_info();
+    }
+    else if (strcmp(command, "exit") == 0)
+    {
+        return true;
+    }
+    else if (strcmp(command, "test") == 0)
+    {
+        run_tests();
+    }
+    else if (strncmp(command, "generate_private", strlen("generate_private") - 1) == 0)
```

Comment:
Why strcmp for everything, but strncmp for this one?

Reply:
I used strncmp for options that might have parameters, that is `generate_private`, `generate_public`, `private_slot`, `public_slot`.